### PR TITLE
Qualified numeric_limits in PFCombinedTauTagInfo.h

### PR DESCRIPTION
**Pull request by Teemperor**

### CODE REVIEW

**Comment by Teemperor** [comment]
Part of the work going on regarding the C++ modules migration of CMSSW (tracked as issue https://github.com/cms-sw/cmssw/issues/15248). This PR is not meant to refactor things, but just to make these headers compile.

**Comment by perrotta** [comment]
@Teemperor : while thanking you for yuor work, can I suggests for your next campaign to merge all such updates  in a single pull requests, in order to reduce the load on the test system (which now needs to run in parallel on several, almost identical pull requests with quite trivial changes) as well as on the reviewers which could gain time in evaluating and signing all them at once?
Thank you,
Andrea

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

File: DataFormats/BTauReco/interface/PFCombinedTauTagInfo.h [hunk 1]
```diff
@@ -36,11 +36,11 @@ namespace reco {
       signedSflightpath_=NAN;
       PFChargedHadrCandsEtJetEtRatio_=NAN;
       PFNeutrHadrCandsE_=NAN;
-      PFNeutrHadrCandsN_=numeric_limits<int>::quiet_NaN();
+      PFNeutrHadrCandsN_=std::numeric_limits<int>::quiet_NaN();
```
Comment (slava77) [inline DataFormats/BTauReco/interface/PFCombinedTauTagInfo.h]: [not asking for a change, just a note]
someone was particularly confusing here: 
- double PFNeutrHadrCandsE_ is assigned an implementation-defined float NAN
- int PFNeutrHadrCandsN_ is assigned a generic programming nan which has standard language value of 0

**Comment by davidlange6** [comment]
merge

**Pull request merged**